src/test/java/echoServer/SocketReaderWriterTest.java

```diff
@@ -29,4 +30,19 @@ void writesALineOfText() throws IOException {
 
         assertEquals("Hello\n", output);
     }
+
+    @Test
+    void itCloses() throws IOException {
+        TestSocket testSocket = new TestSocket("Hello\n");
+        ReadableWriteable socketReaderWriter = new SocketReaderWriter(testSocket);
+
+        socketReaderWriter.readLine();
+        socketReaderWriter.writeLine("Hello");
+
+        socketReaderWriter.close();
+
+        assertThrows(IOException.class, () -> {
```

Comment:
I wanted to prove that the BufferedReader had been closed. The docs say "Once the stream has been closed, further read(), ready(), mark(), reset(), or skip() invocations will throw an IOException." 

Is there a better way to show that the BufferedReader has been closed?

Reply:
Got it. It feels like we are testing implementation details, instead of a behavior. What behavior do we want to test? Or put another way, do we want this test to fail we stop using `BufferedReader`?

Reply:
Ok, makes sense. I want to test that the socket connection is closed. 

The specific behavior I want to test is that you can no longer successfully call readLine or writeLine, but it seems like I get into implementation details when I need to define what a failure to call readLine or writeLine looks like. Maybe it's enough to assert that .close() was called on the socket?

Reply:
I think you are on the right track. You have a `TestSocket` that you have a lot of control over. Perhaps you can use that.